app/scripts/metamask-controller.js

```diff
@@ -8378,6 +8379,21 @@ export default class MetamaskController extends EventEmitter {
     });
   }
 
+  /**
+   * Capture an artificial error in a timeout handler for testing purposes.
+   *
+   * @param message - The error message.
+   * @deprecated This is only mean to facilitiate E2E testing. We should not
+   * use this for handling errors.
+   */
+  captureTestError(message) {
```

Comment:
should these test only functions only be shipped when `process.env.IN_TEST` is `"true"`?

Reply:
They're also used for manual testing, like the other pre-existing ones are. I used them in a `yarn dist` build to discover this issue.

That makes the comment block inaccurate though. I just copied it from the other functions which are used for the exact same purpose. I can fix the comment in a follow-up PR perhaps.

Reply:
Done here: https://github.com/MetaMask/metamask-extension/pull/34404